src/pages/CreateCommunity/CreateCommunityPage.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import { ICustomCommunity, EcommunityType } from 'types/types';
+import { toast } from 'react-toastify';
+import { userInfoAtom } from 'services/recoil/userInfo';
+import { useRecoilValue } from 'recoil';
+import CreateCustomCommunity from 'services/Firebase/WriteData/CommunityUtil';
+import Spinner from 'components/Util/Spinner';
+import {
+  checkCommunityNameValid,
+  checkCommunityInput,
+} from 'functions/InputVerification/CommunityVerfication';
+
+import CommunityDetails from './Components/CommunityDetails';
+import ConnectedTokens from './Components/ConnectedTokens';
+import SocialVerification from './Components/SocialVerification';
+import CommunityManagers from './Components/CommunityManagers';
+
+export default function CreateCommunityPage(): JSX.Element {
+  const userInfo = useRecoilValue(userInfoAtom);
+  const naigate = useNavigate();
+
+  const [community, setCommunity] = useState<ICustomCommunity>({
+    cid: '',
+    type: EcommunityType.custom,
+    name: '',
+    displayName: '',
+    imageUrl: '',
+    tokens: [],
+    requirements: [],
+    managers: [],
+    categories: [],
+  });
+
+  /* Community Details */
+  const [imageFile, setImageFile] = useState<File | Blob | undefined>();
+  const [availabilityMessage, setAvailabilityMessage] = useState(<p> </p>);
+  const [loadingCheckCommunityName, setLoadingCheckCommunityName] =
+    useState(false);
+
+  async function checkCommunityName(
+    userNameBeingChecked: string
+  ): Promise<boolean> {
+    return checkCommunityNameValid(
+      userNameBeingChecked,
+      setLoadingCheckCommunityName,
+      setAvailabilityMessage
+    );
+  }
+
+  /* Tokens requirements */
+
+  const [requirements, setRequirements] = useState<Set<string>>(
+    new Set<string>()
+  );
+
+  /* Submit */
+  const [submitLoading, setSubmitLoading] = useState(false);
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    setSubmitLoading(true);
+    event.preventDefault();
+    const communityBeingChecked = community;
+    communityBeingChecked.managers.unshift(userInfo?.publicKey ?? '');
+    const managersSubmitted = Array.from(
+      new Set(
+        communityBeingChecked.managers.filter((mana) => mana.length === 44)
+      )
+    );
+    communityBeingChecked.managers = managersSubmitted;
+    communityBeingChecked.tokens = Array.from(requirements);
+    communityBeingChecked.requirements = Array.from(requirements);
```

Comment:
Since tokens and requirements are the same, do we need both?

Reply:
Same here. If you're approach is to keep unnecessary complexity at a minimum then let's remove unnecessary requirements field and only add it when we need to.